### PR TITLE
Sc 112776/update date range picker

### PR DESCRIPTION
**Pull request by ajlu**
## Description

- Fix types for `<DatePicker />` and `<DateRangePicker />`
- Add support for no `minDate`
- Add `calendarDay` to theme (Note: `colors.brand.secondary` is the same as `colors.blue`)
- Add typescript reminder in PR checklist

💁‍

## Related issues

https://app.shortcut.com/qantasaccommodation/story/112776/dates

💪

## Looks like this

Screen Shot 2023-01-31 at 10 01 12 am

🤨

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan), [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [x] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

File: src/components/DatePicker/components/CalendarDay/CalendarDay.jsx [hunk 1]
```diff
@@ -62,11 +62,11 @@ const Button = styled(NakedButton, omitProps(['selected']))`
     props.selected &&
     !props.highlighted &&
     css`
-      background-color: ${themeGet('colors.lightBlue')(props)};
+      background-color: ${themeGet('calendarDay.highlightedColor')(props)};
```
Comment (yiyang-qantashotels) [inline src/components/DatePicker/components/CalendarDay/CalendarDay.jsx:65]: so **!props.highlighted** will set background-colour to the highlighted color? 🤔

Comment (ajlu) [reply src/components/DatePicker/components/CalendarDay/CalendarDay.jsx:65]: yeah the original css is a bit confusing, it's more for the next conditional css from line 74 - `props.highlighted && !props.selected` which is shown in the screenshot above. This reuses the same colour

Comment (yiyang-qantashotels) [reply src/components/DatePicker/components/CalendarDay/CalendarDay.jsx:65]: 🤣

**Review by yiyang-qantashotels — APPROVED** [review]

**Pull request merged**